conda_smithy/templates/drone.yml.tmpl

```diff
@@ -0,0 +1,30 @@
+{% if configs[0] -%}
+{%- for config_name, platform, upload, config in configs | sort -%}
+---
+kind: pipeline
+name: {{ config_name }}
+
+platform:
+  os: linux
+  arch: arm64
+
+steps:
+- name: Install and build
+  image: {{ config["docker_image"][-1] }}
+  environment:
+    CONFIG: {{ config_name }}
+    UPLOAD_PACKAGES: {{ upload }}
+    PLATFORM: {{ platform }}
+    #BINSTAR_TOKEN: ""
+  commands:
+  {%- if 'linux' in platform %}
+    - export FEEDSTOCK_ROOT="$CI_WORKSPACE"
+    - export RECIPE_ROOT="$FEEDSTOCK_ROOT/recipe"
+    - export CI=drone
+    - . /opt/conda/bin/activate
```

Comment:
Can you do, `. /opt/docker/bin/entrypoint_source`? It's equivalent to calling activate for now, but it may change

Reply:
Drone drops us into `sh`, so when I do it that way the build errors out.

`
sh: conda: command not found
`

I did a little bit of testing but couldn't find a good way around that, including trying to drop into a bash shell first.  I dunno, maybe there's some other trick?

Reply:
I think running `/opt/docker/bin/entrypoint` will be needed as Drone does not use the entrypoint defined by the container.

Reply:
As far as I can tell `. /opt/conda/bin/activate` gets the minimum needed for the build.

Reply:
`. /opt/conda/bin/activate` will activate the base conda environment but the build will be done by the `root` user.  The container [entrypoint](https://github.com/conda-forge/docker-images/blob/master/scripts/entrypoint) adds a conda user and executes the other build steps as that user.

Reply:
Shouldn't need to do any of that on Drone because we're not using volume mounts, right?

Reply:
It's good to run without sudo to make sure the package doesn't install files outside of prefix. Also, some packages refuse to build under root

Reply:
Hrm ok, probably gonna need some help here then because I don't know how to get it to work correctly with how Drone does the entrypoint and I don't have a deep enough understanding of what all Conda is doing.

Reply:
Run, `/opt/docker/bin/entrypoint /path/to/build_strps.sh` instead activating and running the build script

Reply:
Thanks, getting this error in my test build, still trying to track it down.

```
+ make_build_number /drone/src /drone/src/recipe /drone/src/.ci_support/linux_aarch64_.yaml
--
202 | + conda build /drone/src/recipe -m /drone/src/.ci_support/linux_aarch64_.yaml --clobber-file /drone/src/.ci_support/clobber_linux_aarch64_.yaml
203 | No numpy version specified in conda_build_config.yaml. Falling back to default numpy value of 1.11
204 | WARNING:conda_build.metadata:No numpy version specified in conda_build_config.yaml. Falling back to default numpy value of 1.11
205 | Traceback (most recent call last):
206 | File "/opt/conda/bin/conda-build", line 11, in <module>
207 | sys.exit(main())
208 | File "/opt/conda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 456, in main
209 | execute(sys.argv[1:])
210 | File "/opt/conda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 447, in execute
211 | verify=args.verify, variants=args.variants)
212 | File "/opt/conda/lib/python3.7/site-packages/conda_build/api.py", line 208, in build
213 | notest=notest, need_source_download=need_source_download, variants=variants)
214 | File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 2292, in build_tree
215 | bypass_env_check=True)
216 | File "/opt/conda/lib/python3.7/site-packages/conda_build/render.py", line 784, in render_recipe
217 | m.config.compute_build_id(m.name(), reset=reset_build_id)
218 | File "/opt/conda/lib/python3.7/site-packages/conda_build/config.py", line 529, in compute_build_id
219 | old_dir = self.work_dir if len(os.listdir(self.work_dir)) > 0 else ""
220 | FileNotFoundError: [Errno 2] No such file or directory: '/drone/src/build_artifacts/work'
```

Reply:
Gonna try creating this directory manually first, will update the PR when I have something working.

Reply:
I *think* I have addressed the entrypoint in my latest commit.

Reply:
Any objections to my workaround?